tests/migration/eviction_strategy.go

```diff
@@ -0,0 +1,722 @@
+				By("selecting a node as the source")
+				sourceNode := libnode.GetAllSchedulableNodes(virtClient).Items[0]
+				libnode.AddLabelToNode(sourceNode.Name, cleanup.TestLabelForNamespace(vmis[0].Namespace), "target")
+
+				By("starting four VMIs on that node")
+				for _, vmi := range vmis {
+					_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Create(context.Background(), vmi)
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				By("waiting until the VMIs are ready")
+				for _, vmi := range vmis {
+					libwait.WaitForSuccessfulVMIStart(vmi,
+						libwait.WithTimeout(180),
+					)
+				}
+
+				By("selecting a node as the target")
+				targetNode := libnode.GetAllSchedulableNodes(virtClient).Items[1]
+				libnode.AddLabelToNode(targetNode.Name, cleanup.TestLabelForNamespace(vmis[0].Namespace), "target")
+
+				By("tainting the source node as non-schedulabele")
+				libnode.Taint(sourceNode.Name, libnode.GetNodeDrainKey(), k8sv1.TaintEffectNoSchedule)
+
+				By("waiting until migration kicks in")
+				Eventually(func() int {
+					migrationList, err := virtClient.VirtualMachineInstanceMigration(k8sv1.NamespaceAll).List(&metav1.ListOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					runningMigrations := filterRunningMigrations(migrationList.Items)
+
+					return len(runningMigrations)
+				}, 2*time.Minute, 1*time.Second).Should(BeNumerically(">", 0))
+
+				By("checking that all VMIs were migrated, and we never see more than two running migrations in parallel")
+				Eventually(func() []string {
+					var nodes []string
+					for _, vmi := range vmis {
+						vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &metav1.GetOptions{})
+						nodes = append(nodes, vmi.Status.NodeName)
+					}
+
+					migrationList, err := virtClient.VirtualMachineInstanceMigration(k8sv1.NamespaceAll).List(&metav1.ListOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					runningMigrations := filterRunningMigrations(migrationList.Items)
+					Expect(len(runningMigrations)).To(BeNumerically("<=", 2))
+
+					return nodes
+				}, 4*time.Minute, 1*time.Second).Should(ConsistOf(
+					targetNode.Name,
+					targetNode.Name,
+					targetNode.Name,
+					targetNode.Name,
+				))
+
+				By("Checking that all migrated VMIs have the new pod IP address on VMI status")
+				for _, vmi := range vmis {
+					Eventually(func() error {
+						newvmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &metav1.GetOptions{})
+						Expect(err).ToNot(HaveOccurred(), "Should successfully get new VMI")
+						vmiPod := tests.GetRunningPodByVirtualMachineInstance(newvmi, newvmi.Namespace)
+						return libnet.ValidateVMIandPodIPMatch(newvmi, vmiPod)
+					}, time.Minute, time.Second).Should(Succeed(), "Should match PodIP with latest VMI Status after migration")
+				}
+			})
+		})
+	})
+
+	Describe("[Serial] with a cluster-wide live-migrate eviction strategy set", Serial, func() {
```

Comment:
and this into `cluster-wide-live-migrate-eviction-strategy.go`

Reply:
See comment above